cmd/bbolt/main.go

```diff
@@ -509,16 +502,16 @@ func (cmd *pageItemCommand) Run(args ...string) error {
 	return nil
 }
 
-// leafPageElement retrieves a leaf page element.
-func (cmd *pageItemCommand) leafPageElement(pageBytes []byte, index uint16) (*guts_cli.LeafPageElement, error) {
-	p := (*guts_cli.Page)(unsafe.Pointer(&pageBytes[0]))
+func (cmd *pageItemCommand) validateLeafPage(pageBytes []byte, index uint16) (*common.Page, error) {
+	p := common.LoadPage(pageBytes)
 	if index >= p.Count() {
-		return nil, fmt.Errorf("leafPageElement: expected item index less than %d, but got %d.", p.Count(), index)
+		return nil, fmt.Errorf("leafPageElement: expected item index less than %d, but got %d", p.Count(), index)
 	}
-	if p.Type() != "leaf" {
-		return nil, fmt.Errorf("leafPageElement: expected page type of 'leaf', but got '%s'", p.Type())
+	if p.Typ() != "leaf" {
```

Comment:
The existing `bbolt` package uses `Typ`, while the `cmd/bbolt` uses `Type`. Eventually I selected the former. I seems not a big deal. If you have big concern on this, then I can update it.

Personally I'd like `Typ`, because the there is an existing [Type interface in reflect](https://github.com/golang/go/blob/6f4d986a5af3245a3c66172e5fab9609652ff219/src/reflect/type.go#L39), and actually each value has a [Type method](https://github.com/golang/go/blob/6f4d986a5af3245a3c66172e5fab9609652ff219/src/reflect/value.go#L2601) as well. It isn't a problem technically, but I tend to use `Typ`.